### PR TITLE
test timeout through user-spec added

### PR DESCRIPTION
**Pull request by haribommi**
1. test timeout value can be paased from user-spec.
2. This is useful for handling larger 4K test streams

Signed-off-by: hbomminx <haribabux.bommineni@intel.com>

### CODE REVIEW

**Comment by haribommi** [comment]
example of timeout in user-spec:
  "TheRedwoods" : dict(
    source = os.path.join(assets, "vp9", "10bit", "TheRedwoods.mkv"),
    format = "P010", width = 3840, height = 1620, frames = 4953, timeout = 720)

**Comment by haribommi** [comment]
@uartie , @FocusLuo and @wangzj0601 - could you please review this patch

File: lib/common.py [hunk 1]
```diff
@@ -97,8 +97,13 @@ def timeout(proc):
   logger("CALL: {} (pid: {})".format(command, proc.pid))
 
   reader = threading.Thread(target = readproc, args = [proc])
-  timer = threading.Timer(get_media().call_timeout, timeout, [proc])
 
+  if get_media().userspec_timeout > 0:
+      timeout_val = get_media().userspec_timeout
+  else:
+      timeout_val = get_media().call_timeout
```
[the remark this answers is not in the document]
Comment (haribommi) [reply lib/common.py]: moved to .slashrc as _get_call_timeout()

File: .slashrc [hunk 2]
```diff
@@ -80,6 +80,7 @@ class MediaPlugin(slash.plugins.PluginInterface):
     self.parallel_metrics = args.parallel_metrics
     self.ctapt = args.ctapt
     self.ctapr = args.ctapr
+    self.userspec_timeout = 0
```
[the remark this answers is not in the document]
Comment (haribommi) [reply .slashrc]: renamed to test_call_timeout and added under above functions

File: lib/parameters.py [hunk 3]
```diff
@@ -359,3 +359,10 @@ def gen_transcode_1to1_parameters(spec, codec, mode):
   keys = ("case")
   params = gen_transcode_1to1_variants(spec, codec, mode)
   return keys, params
+
+def gen_get_test_timeout(test_name, spec):
+  from common import get_media
+  for case, params in spec.iteritems():
+    variant = params.get("timeout", None)
```
[the remark this answers is not in the document]
Comment (haribommi) [reply lib/parameters.py]: changed to call_timeout

File: lib/parameters.py [hunk 4]
```diff
@@ -359,3 +359,10 @@ def gen_transcode_1to1_parameters(spec, codec, mode):
   keys = ("case")
   params = gen_transcode_1to1_variants(spec, codec, mode)
   return keys, params
+
+def gen_get_test_timeout(test_name, spec):
```
[the remark this answers is not in the document]
Comment (haribommi) [reply lib/parameters.py]: removed this function

File: test/ffmpeg-qsv/decode/10bit/vp9.py [hunk 5]
```diff
@@ -26,6 +26,7 @@ def test(self, case):
       ffdecoder = "vp9_qsv",
       hwformat  = "p010le",
     )
+    gen_get_test_timeout(case, spec)
```
[the remark this answers is not in the document]
Comment (haribommi) [reply test/ffmpeg-qsv/decode/10bit/vp9.py]: added vars(self).get("call_timeout", 0) under decode and transcode base class.

**Comment by haribommi** [comment]
> IIRC, we mostly needed this feature for some of the transcode cases, yet I don't see any modification to transcode tests.

added user-spec timeout reading under transcode tests and will add in config for required streams.

File: .slashrc [hunk 6]
```diff
@@ -262,6 +264,11 @@ class MediaPlugin(slash.plugins.PluginInterface):
     filename = os.path.join(
       slash.context.session.results.global_result.get_log_dir(), "results.xml")
     tree.write(filename)
+  def _get_call_timeout(self):
```
Comment (uartie) [inline .slashrc]: Please move this towards the top of the class where other functions that start with `_` are located.  Also, please make sure you have an empty line between function definitions.

Comment (haribommi) [reply .slashrc]: moved _get_call_timeout to top where other _ are defined

File: lib/common.py [hunk 7]
```diff
@@ -97,8 +97,7 @@ def timeout(proc):
   logger("CALL: {} (pid: {})".format(command, proc.pid))
 
   reader = threading.Thread(target = readproc, args = [proc])
-  timer = threading.Timer(get_media().call_timeout, timeout, [proc])
-
+  timer = threading.Timer(get_media()._get_call_timeout(), timeout, [proc])
```
Comment (uartie) [inline lib/common.py]: Sorry, I just noticed this... there is another get_media().call_timeout usage on line 121 that needs updated too.

Comment (uartie) [reply lib/common.py]: Maybe on line 121, we should use timer.interval instead of get_media().call_timeout.

Comment (haribommi) [reply lib/common.py]: under line 121, changed get_media().call_timeout to timer.interval

Comment (uartie) [reply lib/common.py]: I don't see the change.  Did you forget to push?

Comment (haribommi) [reply lib/common.py]: sorry, my mistake can you check now

**Pull request merged**